File: backend/app/schemas/public.py
from pydantic import BaseModel

from app.schemas.stats import BadgeRead, OverviewStats, StreakStats


class PublicProfile(BaseModel):
    username: str
    display_name: str | None
    overview: OverviewStats
    streaks: StreakStats
    badges: list[BadgeRead]

File: frontend/src/api/public.ts
import { api } from "../lib/apiClient";
import type { Badge as BadgeType, OverviewStats, StreakStats } from "./stats";

export interface PublicProfile {
  username: string;
  display_name: string | null;
  overview: OverviewStats;
  streaks: StreakStats;
  badges: BadgeType[];
}

export const publicApi = {
  profile: (shareToken: string) => api.get<PublicProfile>(`/public/${shareToken}`),
};
